File: pset1/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main()
{
    int height = 0;

    do
    {
        height = get_int("Let's build a pyramide. Give me a number between 1 and 8 inclusive! ");

        if (height > 0 && height < 9)
        {
            int currentRow = 1;

            while (currentRow <= height)
            {
                for (int i = 0; i < height - currentRow; i++) // prints spaces on left side
                {
                    printf(" ");
                }

                for (int j = 0; j < currentRow; j++) // prints left #
                {
                    printf("#");
                }

                printf("  "); // spaces between pyramid

                for (int k = 0; k < currentRow; k++) // prints right #
                {
                    printf("#");
                }

                printf("\n");

                currentRow++;
            }

            return 0;
        }
    }
    while (height < 1 || height > 8);
}

File: pset6/mario-more/mario.py
MIN_HEIGHT = 1
MAX_HEIGHT = 8

# Loop for user input
while True:
    try:
        # Prompt the user for height
        height = int(input("Height: "))

        # If the height is in the allowed range
        if MIN_HEIGHT <= height <= MAX_HEIGHT:
            break

    except ValueError:
        # Ignore invalid inputs and prompt again
        pass

# Iterate through each row
for i in range(1, height + 1):
    # Print spaces at the beginning of the row
    print(" " * (height - i), end="")

    # Print the left-hand side of the pyramid
    print("#" * i, end="")

    # Print the gap between the two sides
    print("  ", end="")

    # Print right-hand side of the pyramid
    print("#" * i)
